src/uct/base/uct_cm.c

```diff
@@ -82,28 +82,83 @@ ucs_status_t uct_cm_ep_pack_cb(uct_cm_base_ep_t *cep, void *arg,
     return status;
 }
 
+ucs_status_t uct_cm_ep_client_set_connect_cb(const uct_ep_params_t *params,
+                                             uct_cm_base_ep_t *cep)
+{
+    if (params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_CONNECT_CB) {
+        if (params->sockaddr_connect_cb.client != NULL) {
+            cep->client.connect_cb = params->sockaddr_connect_cb.client;
+        } else {
+            ucs_error("UCT_EP_PARAM_FIELD_SOCKADDR_CONNECT_CB is set but client's"
+                      "connect callback is NULL");
+            return UCS_ERR_INVALID_PARAM;
+        }
+    } else {
+        cep->client.connect_cb = NULL;
+    }
+
+    return UCS_OK;
```

Comment:
@yosefe this means to put the 'unlikely to happen' code first (perf wise speaking). 
is flattening the code here more important since it isn't the fast path?

Reply:
we should flatten the code, and for fast path case, add"ucs_unlikely"